platform/core/src/services/UIModalService/index.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * A UI Element
+ *
+ * @typedef {HTMLElement} Modal
+ */
+
+/**
+ * UI Modal
+ *
+ * @typedef {Object} ModalProps
+ * @property {string} [backdrop=false] -
+ * @property {string} [keyboard=false] -
+ * @property {number} [show=true] -
+ * @property {string} [closeButton=true] -
+ * @property {string} [title=null] - 'Modal Title'
+ * @property {boolean} [customClassName=null] - '.ModalClass'
+ */
+
+const uiModalServicePublicApi = {
+  name: 'UIModalService',
+  hide,
+  show,
+  setServiceImplementation,
+};
+
+const uiModalServiceImplementation = {
+  _hide: () => console.warn('hide() NOT IMPLEMENTED'),
+  _show: () => console.warn('show() NOT IMPLEMENTED'),
+};
+
+function createUiModalService() {
+  return uiModalServicePublicApi;
+}
+
+/**
+ * Show a new UI modal;
+ *
+ * @param {Modal} component
+ * @param {ModalProps} props { backdrop, keyboard, show, closeButton, title, customClassName }
+ */
```

Comment:
`component` - Probably a comment that indicates this should be a React component (as redundant as that may seem)

`ModalProps` - Neat use of typdef

Reply:
Done! thank you.